gunrock/app/sm/sm_app.cuh

```diff
@@ -132,17 +133,21 @@ cudaError_t RunTests(util::Parameters &parameters, GraphT &data_graph,
 
 /*
  * @brief Entry of gunrock_sm function
- * @tparam     GraphT     Type of the graph
- * @tparam     ValueT     Type of the distances
- * @param[in]  parameters Excution parameters
- * @param[in]  graph      Input graph
- * @param[out] distances  Return shortest distance to source per vertex
- * @param[out] preds      Return predecessors of each vertex
- * \return     double     Return accumulated elapsed times for all runs
+ * @tparam     GraphT          Type of the graph
+ * @tparam     ValueT          Type of the distances
+ * @param[in]  parameters      Excution parameters
+ * @param[in]  data_graph      Input data graph
+ * @param[in]  query_graph     Input query graph
+ * @param[out] subgraphs       Return number of matched subgraphs
+ * @param[out] list_subgraphs  Return list of matched subgraph combinations
+ * @param[in]  device          Target device where inputs and outputs are stored
+ * \return     double          Return accumulated elapsed times for all runs
  */
 template <typename GraphT, typename ValueT = typename GraphT::ValueT>
 double gunrock_sm(gunrock::util::Parameters &parameters, GraphT &data_graph,
-                  GraphT &query_graph, typename GraphT::VertexT *subgraphs) {
+                  GraphT &query_graph, unsigned long *subgraphs,
+                  unsigned long *list_subgraphs,
+                  gunrock::util::Location device = gunrock::util::HOST) {
```

Comment:
Naming convention here as well, maybe change it to `allocated_on`?

Reply:
Sure